File: Meraki_practice/palindrome_checker.rb
#!/usr/bin/ruby

class PalindromeChecker
	@file_path=nil;@text=nil;
	def initialize(params={})
		@file_path = params["-f"] if params["-f"] != nil and params["--text"] == nil;
		@text = params["--text"].strip if params["--text"] != nil and params["-f"] == nil;
	end
	def read_file()
		if (File.exist?("#{@file_path}"))
			File.open(@file_path, 'r') do |f|
				f.each_line do |line|
					@text += "#{line}";
				end
			end
		else
			throw "File does not exist...";
		end
	end
	def check_palindrome(text)
		for i in 0..text.length - 1
			if (text[i] != text[text.length - i - 1])
				return false;
			end
		end
		return true;
	end
	def invoke()
		if (@file_path != nil)
			@text = "";
			self.read_file();
		end
		@text.split("\n").each do |l| print "#{l} => #{self.check_palindrome(l)}\n"; end;
	end
end

params = {}
for i in 0..ARGV.length
	params[ARGV[i]] = ARGV[i + 1];
end

session = PalindromeChecker.new(params);
session.invoke();

File: Meraki_practice/palindrome_checker.py
#!/usr/bin/env python3
###############################################################################
# Name        : palindrome_checker.py                                         #
# Author      : Abel Gancsos                                                  #
# Version     : v. 1.0.0.0                                                    #
# Description : Practice working with palindromes.                            #
###############################################################################
import os, sys;

class PalindromeChecker:
	file_path=None;text=None;
	def __init__(self, params=dict()):
		sorted_dict = sorted(params.items(), key=lambda x:x[1]);
		print(sorted_dict);
		self.text = params["--text"] if "--text" in params.keys() else "";
		self.file_path = params["-f"] if "-f" in params.keys() else "";
		if (self.file_path != ""):
			with open(self.file_path, 'r') as fh: self.text = fh.read();
		pass;
	def check_palindrome(self, text):
		for i in range(0, len(text) - 1):
			if (text[i] != text[len(text) - i - 1]): return False;
		return True;
	def invoke(self):
		for line in self.text.split("\n"): 
			if (line == ""): continue;
			print("{0} => {1}".format(line, self.check_palindrome(line)));
		pass;
	pass;

if __name__ == "__main__":
	params = dict();
	for i in range(0, len(sys.argv) - 1): params[sys.argv[i]] = sys.argv[i + 1];
	session = PalindromeChecker(params);
	session.invoke();
	pass;

